### ISSUE TITLE
IfcCurveSegment - missing Derive: Dim

### ISSUE THREAD

**Issue by pjanck** [bSI-InfraRoom/IFC-Specification#7]
`IfcCurveSegment` needs a derive attribute `Dim` in order for the `IfcCurveDim` function to work as expected. See snippets below:

- `IfcCompositeCurveSegment` has a derived `Dim`:

https://github.com/bSI-InfraRoom/IFC-Specification/blob/27075c5519f85d83db5d76886b70ac708b3745d1/IFC4x3_RC3.exp#L5131-L5141

- `IfcCurveSegment` on the other hand does not:

https://github.com/bSI-InfraRoom/IFC-Specification/blob/27075c5519f85d83db5d76886b70ac708b3745d1/IFC4x3_RC3.exp#L5649-L5655

- `IfcCurve` gets its `Dim` using the `IfcCurveDim` function:

https://github.com/bSI-InfraRoom/IFC-Specification/blob/27075c5519f85d83db5d76886b70ac708b3745d1/IFC4x3_RC3.exp#L5620-L5633

- Inside the `IfcCurveDim` function, the composite curve gets handled as "returning the dim of its first segment":

https://github.com/bSI-InfraRoom/IFC-Specification/blob/27075c5519f85d83db5d76886b70ac708b3745d1/IFC4x3_RC3.exp#L12857-L12859

Well, this fails, if the segment is of type `IfcCurveSegment`.

**Comment by SergejMuhic** [comment]
TL;DR

This is what I did:

ENTITY IfcCurveSegment
 SUBTYPE OF (IfcSegment);
	Placement : IfcPlacement;
	SegmentStart : IfcCurveMeasureSelect;
	SegmentLength : IfcCurveMeasureSelect;
	ParentCurve : IfcCurve;
 DERIVE
	Dim : IfcDimensionCount := ParentCurve.Dim;
END_ENTITY;

**Comment by pjanck** [comment]
I consider this closed then? Have you made a pull request that closes the issue?

**Comment by SergejMuhic** [comment]
and this because segments can be 2d
FUNCTION IfcCurveDim
(Curve : IfcCurve)
       : IfcDimensionCount;

  IF ('IFC4X3_RC3.IFCLINE' IN TYPEOF(Curve))
    THEN RETURN(Curve\IfcLine.Pnt.Dim);
  END_IF;
  IF ('IFC4X3_RC3.IFCCONIC' IN TYPEOF(Curve))
    THEN RETURN(Curve\IfcConic.Position.Dim);
  END_IF;
  IF ('IFC4X3_RC3.IFCPOLYLINE' IN TYPEOF(Curve))
    THEN RETURN(Curve\IfcPolyline.Points[1].Dim);
  END_IF;
  IF ('IFC4X3_RC3.IFCTRIMMEDCURVE' IN TYPEOF(Curve))
    THEN RETURN(IfcCurveDim(Curve\IfcTrimmedCurve.BasisCurve));
  END_IF;
  IF ('IFC4X3_RC3.IFCGRADIENTCURVE' IN TYPEOF(Curve))
    THEN RETURN(3);
  END_IF;
  IF ('IFC4X3_RC3.IFCSEGMENTEDREFERENCECURVE' IN TYPEOF(Curve))
    THEN RETURN(3);
  END_IF;
  IF ('IFC4X3_RC3.IFCCOMPOSITECURVE' IN TYPEOF(Curve))
    THEN RETURN(Curve\IfcCompositeCurve.Segments[1].Dim);
  END_IF;

**Comment by pjanck** [comment]
Good work! Closing.
